### PR TITLE
Fix kubectl logs for init containers

### PR DESCRIPTION
**Pull request by lukaszo**
Related issues: #25818 #27040

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -2845,6 +2845,10 @@ func (kl *Kubelet) validateContainerLogStatus(podName string, podStatus *api.Pod
 	var cID string
 
 	cStatus, found := api.GetContainerStatus(podStatus.ContainerStatuses, containerName)
+	// if not found, check the init containers
```
Comment (dims) [inline pkg/kubelet/kubelet.go]: @lukaszo Trying to understand, this covers only the case where the container name is specified in the command line. right? (and not the problems mentioned in #27040)

Comment (dchen1107) [inline pkg/kubelet/kubelet.go]: This one fixes the issue of pulling init container's log. Not addressed #27040.

**Comment by dchen1107** [comment]
LGTM

cc/ @smarterclayton

**Comment by smarterclayton** [comment]
Also Lgtm, thanks

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (lukaszo) [inline pkg/kubelet/kubelet.go]: @dims yes, exactly. Working on #27040 without this doesn't make sense

Comment (krmayankk) [inline pkg/kubelet/kubelet.go]: @dchen1107 what is a init container logs

Comment (lukaszo) [inline pkg/kubelet/kubelet.go]: @krmayankk see here https://github.com/kubernetes/kubernetes/pull/23666

**Comment by dchen1107** [comment]
@k8s-bot test this github issue: #27276

**Comment by dchen1107** [comment]
The failure has nothing to do with this pr. Reapply lgtm

**Comment by krmayankk** [comment]
@lukaszo thanks for the pointer, this feature looks very interesting and usable. Can someone also point to examples of how to use it ? I very often see the PR for new features get merged without an example in the docs/user guides or contrib. It would help to make that as requirement for these new PR merges imho

**Comment by lukaszo** [comment]
@krmayankk init containers are for example used in petsets https://github.com/kubernetes/contrib/blob/master/pets/mysql/galera/mysql-galera.yaml#L32-L76

**Comment by smarterclayton** [comment]
We definitely want people to try alpha features, but they're also somewhat alpha and subject to change.  We'll probably have some core documentation for petset but I would expect them to change in 1.4 as we learn more about what we need to implement.

**Comment by idvoretskyi** [comment]
@smarterclayton so, are we going to move this feature to 1.4 milestone?

**Comment by smarterclayton** [comment]
Not sure what you're asking?  I had no plans to move this PR out of v1.3 since it's contained, reviewed, and has been in the queue for a few days.

**Comment by lukaszo** [comment]
@smarterclayton are P3 PRs still merged?

**Comment by smarterclayton** [comment]
Anything in the queue is being left in the queue - we may decide to cut the RC without everything in the queue being merged but that decision hasn't been made.  

This also probably isn't a P3.

**Comment by smarterclayton** [comment]
This is P1 because it would block debugging of pet sets for end users.  We can kick it to P2 if necessary, but I'd be more concerned without this than with other PRs related to init containers.

**Pull request merged**